app/controllers/api/v2/instance_hosts_controller.rb

```diff
@@ -0,0 +1,32 @@
+module Api
+  module V2
+    class InstanceHostsController < V2::BaseController
+      def resource_class
+        @resource_class ||= Host::Managed
+      end
+
+      def resource_scope(*args)
+        super.authorized(:view_hosts).joins(:infrastructure_facet).merge(::HostFacets::InfrastructureFacet.where(foreman: true))
```

Comment:
Since the facet is infrastructure, I wondered a bit if we should root the nomenclature in Infrastructure everywhere. I think that would translate then from InstanceHosts to having Foreman Infrastructure Hosts, and Smart Proxy Infrastructure Hosts. I was thinking about this too from looking at the code where internally we reference fields and call them `foreman` but externally we present this as `instance`. The model says, an infrastructure facet can have a smart_proxy and a foreman. The API would say a host can have an instance and a smart_proxy. So we have to connect an instance is a Foreman. 

Maybe just call it `foreman` and accept the weirdness at the start but get the commonality. Or `foreman_instance` to be doubly clear and then use that everywhere else.

Just some food for thought to reduce the mental translation when reading the code in the future for developers.

Reply:
I agree that the naming is currently a bit unfortunate.

> The model says, an infrastructure facet can have a smart_proxy and a foreman.

To be precise, host can *have* a smart proxy and *be* foreman. I'm sorry if I'm being pedantic, but the host can't really *have* a foreman since there's no object that could be on the other end of the association. 
/derail

> Or foreman_instance to be doubly clear and then use that everywhere else.

Currently `foreman` field is a boolean. Wouldn't it be a bit surprising if we renamed it to `foreman_instance`. Wouldn't then people expect it returning an actual instance?

Reply:
Renamed the field to `foreman_instance`